powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/internal/ValidationAspect.java

```diff
@@ -59,7 +64,9 @@ && placedOnRequestHandler(pjp)) {
                 JsonSchema inboundJsonSchema = getJsonSchema(validation.inboundSchema(), true);
 
                 Object obj = pjp.getArgs()[0];
-                if (obj instanceof APIGatewayProxyRequestEvent) {
+                if (validation.envelope() != null && !validation.envelope().isEmpty()) {
```

Comment:
Could there be cases where customer functionality changes because of this reordering?